### PR TITLE
Add doc to scraper module

### PR DESCRIPTION
**Pull request by rathrio**

### CODE REVIEW

File: app/scrapers/scraper.rb [hunk 1]
```diff
@@ -1,4 +1,7 @@
+# Include this module to a class to make it behave like a scraper.
 module Scraper
```
Comment (rathrio) [inline app/scrapers/scraper.rb]: TODO: add small example

**Comment by rathrio** [comment]
@simplay what do you think?

File: app/scrapers/scraper.rb [hunk 2]
```diff
@@ -26,6 +34,9 @@ def scrape
     end
   end
 
+  # @return [Array] an Array of objects most likely retrieved via an API call.
+  #   Each element will get passed to the scraper template methods in order to
```
Comment (simplay) [inline app/scrapers/scraper.rb:69]: @rathrio: Maybe we should elaborate what is actually meant by _templete methods_ .

File: app/scrapers/scraper.rb [hunk 3]
```diff
@@ -26,6 +34,9 @@ def scrape
...
+  #   extract the desired data.
```
Comment (simplay) [inline app/scrapers/scraper.rb:70]: Maybe we should mention here that the returned object is supposed to implements `#map`.

File: app/scrapers/scraper.rb (hunk 1 above)
Comment (simplay) [reply app/scrapers/scraper.rb]:
```
# Steps:
#   1. Define a new scraper class that includes the `Scraper` module.
#   2. Define the `search_results` method.
#   3. Define your tempalte scraper methods.
#
# @example:
# 
#class BoringScraper
#  include Scraper
#  
#  # In general we rather would fetch some conetent from a client implementation.
#  def search_results
#    [{name: "foo"}, {name: "bar"}]
#  end
#  
#  # An example of a template scraper method
#  def scrape_name(result)
#   result[:name]
#  end
#end
```

Comment (rathrio) [reply app/scrapers/scraper.rb]: Exactly like this 👍 😄

File: app/scrapers/scraper.rb (hunk 2 above)
Comment (rathrio) [reply app/scrapers/scraper.rb:69]: Yeah, good idea. Will do.

File: app/scrapers/scraper.rb (hunk 3 above)
Comment (rathrio) [reply app/scrapers/scraper.rb:70]: I don't think that's necessary. It's implied with the return type `Array`, and that we use`map` is more of an implementation detail.

**Comment by simplay** [comment]
@rathrio: Thanks a lot for documenting this class that well - well done!

**Pull request merged**